packages/core/src/components/TextArea/TextArea.tsx

```diff
@@ -34,12 +37,19 @@ const TextArea = forwardRef(
     const numRows = rows || DEFAULT_ROWS[size];
     const helpTextId = helpText && `${id}-help-text`;
 
+    const [characterCount, setCharacterCount] = useState(rest.value?.length || 0);
+
+    const handleOnChange = useCallback((event: React.ChangeEvent<HTMLTextAreaElement>) => {
+      setCharacterCount(event.target.value.length);
+      onChange?.(event);
+    }, [onChange]);
+
     return (
       <div
         className={cx(
           styles.textAreaWrapper,
           {
-            [styles.error]: error,
+            [styles.error]: error || (maxLength && characterCount > maxLength),
```

Comment:
Should we show an error? I think the component should just block new inputs.

Reply:
Once our character count exceeds the maxLength (which only happens when `allowExceedingLimit` is set to true) we display the input in an error state. Pulled this styling from the Figma designs.